src/mlpack/bindings/julia/julia_option.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file julia_option.hpp
+ * @author Ryan Curtin
+ *
+ * The Julia option type.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_BINDINGS_PYTHON_JULIA_OPTION_HPP
+#define MLPACK_BINDINGS_PYTHON_JULIA_OPTION_HPP
+
+#include <mlpack/core/util/param_data.hpp>
+#include "get_param.hpp"
+#include "get_printable_param.hpp"
+#include "print_param_defn.hpp"
+#include "print_input_param.hpp"
+#include "print_input_processing.hpp"
+#include "print_output_processing.hpp"
+#include "print_doc.hpp"
+#include "default_param.hpp"
+
+namespace mlpack {
+namespace bindings {
+namespace julia {
+
+// Defined in mlpack_main.hpp.
+extern std::string programName;
+
+/**
+ * The Python option class.
+ */
+template<typename T>
+class JuliaOption
+{
+ public:
+  /**
+   * Construct a JuliaOption object.  When constructed, it will register itself
+   * with CLI. The testName parameter is not used and added for compatibility
+   * reasons.
+   */
+  JuliaOption(const T defaultValue,
+              const std::string& identifier,
+              const std::string& description,
+              const std::string& alias,
+              const std::string& cppName,
+              const bool required = false,
+              const bool input = true,
+              const bool noTranspose = false,
+              const std::string& /* testName */ = "")
+  {
+    // Create the ParamData object to give to CLI.
+    util::ParamData data;
+
+    data.desc = description;
+    data.name = identifier;
+    data.tname = TYPENAME(T);
+    data.alias = alias[0];
+    data.wasPassed = false;
+    data.noTranspose = noTranspose;
+    data.required = required;
+    data.input = input;
+    data.loaded = false;
+
+    // Only "verbose" will be persistent.
+    if (identifier == "verbose")
+      data.persistent = true;
+    else
+      data.persistent = false;
+    data.cppType = cppName;
+
+    // Every parameter we'll get from Python will have the correct type.
```

Comment:
Do you mean Julia?

Reply:
Right, there was a good amount of code that I based on the Python implementation.  I went through and changed any references from Python to Julia as needed in eee1173a.